### DISCUSSION TITLE
Deployer cannot handle magento 2 static content deploy with multiple jobs

### DISCUSSION THREAD

**Discussion by arditza** [deployphp/deployer#3897] [Bugs]
### Deployer Version

v7.3.3

### Target OS

CentOS Linux 7 (Core)

### Which PHP version are you using?

PHP 8.1

### Content of deploy.php or deploy.yaml

```php
<?php

namespace Deployer;

require_once 'recipe/magento2.php';

// CONFIG
set('update_code_strategy', 'clone');
set('git_ssh_command', 'ssh -o StrictHostKeyChecking=no');
set('ssh_arguments', ['-o UserKnownHostsFile=/dev/null', '-o StrictHostKeyChecking=no', '-o ForwardAgent=yes']);
set('port', 22);

set('env', [
    'DO_NOT_TRACK' => 'true',
    'DO_NOT_SHOW_BANNER' => 'true',
]);

// Project name
set('application', 'deployer-magento2');

// Increase the default timeout of 10000
set('default_timeout', 10000);

// Deploy without setting maintenance mode if possible
set('enable_zerodowntime', false);

// reindex if necessary
set('enable_reindex', false);

// deploy static content per separate language
set('separate_content_language_deploy', false);

// always run setup upgrade as this seems to not work properly
set('database_upgrade_needed', true);

// Use composer installed by Synoa
set('bin/composer', "{{bin/php}} ~/bin/composer");

set('writable_use_sudo', false);

set('keep_releases', 3);

add('shared_files', [
    'auth.json'
]);

// deploy.php 
set('static_content_locales', 'de_DE en_US en_GB fr_FR it_IT pl_PL es_ES');
set('separate_content_language_deploy', false);
set('bin/php', 'php -d allow_url_fopen=on -d memory_limit=8G');
set('static_content_jobs', '3 -vv');

host('prd')
    ->set('hostname', $hostPrd)
    ->set('remote_user', $userPrd)
    ->set('branch', $branchPrd)
    ->set('deploy_path', "/home/$userPrd/magento2")
    ->set('static_content_jobs', 3)
    ->set('repository', $repo);
```
    

### Steps to reproduce

Magento version: 2.4.5-p5
php: 8.1

Applied magento patch for the static content deployment https://github.com/magento/magento2/commit/12a9eacaf57baf1dbd71b22cbd9ada32f3ffa479 

When deploying a magento project with the magento2 recipe, the static content deployment `magento:deploy:assets` fails when using multiple jobs `php -d allow_url_fopen=on -d memory_limit=8G bin/magento setup:static-content:deploy --content-version=123456789  de_DE  -j 3 --exclude-theme Magento/luma` with the following error message: **exit code 141 (Write to pipe with no one reading)** . 

Everything works ok when removing the jobs parameter `-j 3`. Also the above command `php -d allow_url_fopen=on -d memory_limit=8G bin/magento setup:static-content:deploy --content-version=123456789  de_DE  -j 3 --exclude-theme Magento/luma` works properly when running it directly on the server when connected via ssh. 

To me it looks like the deployer recipe for some reason cannot handle multiple jobs on the static content deployment or i am missing some sort of configuration which i cannot figure out. 

Any feedback would be very much appreciated

**Comment by antonmedv** [comment]
@peterjaap please, take a look.

**Reply by peterjaap** [reply]
@antonmedv pretty sure it doesn't have anything to do with Magento specifically. Any chance there's something in Deployer that has issues with spawning multiple PHP processes and getting the output from these processes? That's what the error 141 / SIGPIPE looks like to me.

**Comment by antonmedv** [comment]
I see. Will take a look
